ruby/README.md

```diff
@@ -0,0 +1,211 @@
+```
+* Min threads: 5, max threads: 5
+* Environment: development
+* Listening on tcp://127.0.0.1:5000
+* Listening on tcp://[::1]:5000
+```
+
+To access the app, open http://127.0.0.1:5000/ in your browser.
+
+# Run you first verification
+## Overview
+The Ruby on Rails Quickstart app emulates the experience of an applicant going through a background check/income verification and visiting the applicant portal.
+
+Before using Citadel for verification, applicants filled out the form. 
+
+To streamline the process and make employment/income verification easy and instant, we "hide" the form behind the button. 
+
+If the verification is successful via Citadel, then we show to the applicant the data that we found in their payroll account. 
+
+If the verification isn't successful or the applicant decided to exit Citadel's widget, the applicant will see the form, fill it out and the verification can be done via an existing verification process.
+
+## Successful verification
+
+After opening the Ruby on Rails Quickstart app running locally, click the `Verify employment`/`Verify income` button, search for a company, eg `Facebook` and select any provider. 
+
+Use the Sandbox credentials to simulate a successful login.
+
+```
+username: goodlogin
+password: goodpassword
+```
+
+Once you have entered your credentials and moved to the next screen, you have succesfully done your first verification. 
+
+The API call will be executed and the data will be loaded into the fields of the form.
+
+## No verification
+
+Now click `Add employer` button, search for a company, eg `Facebook` and select any provider. 
+
+Click exit icon at the top right of the widget and you'll see the empty form.
+
+# What happened under the hood
+
+- :smiley: = User
+- :computer: = Front End/Client App
+- :cloud: = Back End/Server
+
+Here is the flow that a successful verification process takes in our example:
+
+1. [:computer: runs `CitadelBridge.init` with `public_key`](#step-1)
+2. [:smiley: clicks `Verify Income/`Verify Employment` button](#step-2)
+3. [:computer: displays Citadel widget, fires `onLoad` function executed](#step-3)
+4. [:smiley: selects employer, choses provider, logs in, clicks `Done`](#step-4)
+5. [:computer: first onSuccess function, sends request to :cloud: with temporary `token`, closes widget, first `onClose`](#step-5)
+6. [:cloud: sends API request to Citadel exchanging temporary `token` for `access_token`](#step-6)
+7. [:cloud: sends API request to Citadel with `access_token` for employment/income verification](#step-7)
+8. [:cloud: sends employment/income verification information back to :computer:](#step-8)
+9. [:computer: renders the verification info sent back by :cloud: for :smiley: to view](#step-9)
+
+## <a id="step-1"></a>1. :computer: runs `CitadelBridge.init` with `public_key`
+```
+  const bridge = CitadelBridge.init({
+    clientName: 'Citadel Quickstart',
+    companyMappingId: null,
+    key: '{{public_key}}',
+    product: 'income',
```

Comment:
employment

Reply:
referenced as `product_type` since instructions are showing for either employment or income